server/datastore/mysql/activities.go

```diff
@@ -19,17 +21,29 @@ func (ds *Datastore) NewActivity(ctx context.Context, user *fleet.User, activity
 
 	var userID *uint
 	var userName *string
+	var userEmail *string
 	if user != nil {
 		userID = &user.ID
 		userName = &user.Name
+		userEmail = &user.Email
 	}
 
-	_, err = ds.writer(ctx).ExecContext(ctx,
-		`INSERT INTO activities (user_id, user_name, activity_type, details) VALUES(?,?,?,?)`,
+	cols := []string{"user_id", "user_name", "activity_type", "details"}
+	insertStmt := `INSERT INTO activities (%s) VALUES (%s)`
+	args := []any{
 		userID,
 		userName,
 		activity.ActivityName(),
 		detailsBytes,
+	}
+	if userEmail != nil {
```

Comment:
Do we actually have users without emails? If so, please add a test for that case.

Reply:
We don't have users without emails, but we do have cases where there is no email when this code is running (e.g. a failed login attempt). There's an existing test for failed logins which would result in this conditional being executed